skimage/measure/_colocalization.py

```diff
@@ -0,0 +1,371 @@
+        Only `imgA` and `imgB` pixel values within this region of interest mask
+        are included in the calculation.
+        Must have same dimensions as `imgA`.
+
+    Returns
+    -------
+    moc: float
+        Manders' Overlap Coefficient of pixel intensities between the two
+        images.
+
+    Notes
+    -------
+    Manders' Overlap Coefficient (MOC) is given by the equation [1]_:
+
+    .. math::
+        r = \frac{\sum A_i B_i}
+                 {\sqrt{\sum A_i^2 \sum B_i^2}}
+
+    where
+        :math:`A_i` is the value of the :math:`i^{th}` pixel in `imgA`
+        :math:`B_i` is the value of the :math:`i^{th}` pixel in `imgB`
+
+    It ranges between 0 for no colocalization and 1 for complete colocalization
+    of all pixels.
+
+    MOC does not take into account pixel intensities, just the fraction of
+    pixels that have positive values for both channels[2]_ [3]_. Its usefulness
+    has been criticized as it changes in response to differences in both
+    co-occurence and correlation and so a particular MOC value could indicate
+    a wide range of colocalization patterns [4]_ [5]_.
+
+    References
+    ----------
+    .. [1] Manders, E.M.M., Verbeek, F.J. and Aten, J.A. (1993), Measurement of
+           co-localization of objects in dual-colour confocal images. Journal
+           of Microscopy, 169: 375-382.
+           https://doi.org/10.1111/j.1365-2818.1993.tb03313.x
+           https://imagej.net/media/manders.pdf
+    .. [2] Dunn, K. W., Kamocka, M. M., & McDonald, J. H. (2011). A practical
+           guide to evaluating colocalization in biological microscopy.
+           American journal of physiology. Cell physiology, 300(4), C723–C742.
+           https://doi.org/10.1152/ajpcell.00462.2010
+    .. [3] Bolte, S. and Cordelières, F.P. (2006), A guided tour into
+           subcellular colocalization analysis in light microscopy. Journal of
+           Microscopy, 224: 213-232.
+           https://doi.org/10.1111/j.1365-2818.2006.01
+    .. [4] Adler J, Parmryd I. (2010), Quantifying colocalization by
+           correlation: the Pearson correlation coefficient is
+           superior to the Mander's overlap coefficient. Cytometry A.
+           Aug;77(8):733-42.https://doi.org/10.1002/cyto.a.20896
+    .. [5] Adler, J, Parmryd, I. Quantifying colocalization: The case for
+           discarding the Manders overlap coefficient. Cytometry. 2021; 99:
+           910– 920. https://doi.org/10.1002/cyto.a.24336
+
+    """
+    if roi is None:
+        roi = np.ones_like(imgA)
+    check_numpy_arr(imgA, 'imgA', bool_expected=False)
+    check_numpy_arr(imgB, 'imgB', bool_expected=False)
+    check_numpy_arr(roi, 'roi', bool_expected=True)
+    check_shape_equality_all(imgA, imgB, roi)
+
+    imgA = imgA[roi.astype(bool)]
+    imgB = imgB[roi.astype(bool)]
+
+    denom = (np.sum(np.square(imgA)) * (np.sum(np.square(imgB)))) ** 0.5
+    return np.sum(np.multiply(imgA, imgB)) / denom
```

Comment:
To avoid overhead, lets not generate `roi` in cases where it is not needed. 

```suggestion

image0 = np.asarray(image0)
image1 = np.asarray(image1)
if roi:
    roi = np.asarray(roi, dtype=bool)
    check_shape_equality_all(image0, image1, roi)
    image0 = image0[roi]
    image1 = image1[roi]
else:
    check_shape_equality_all(image0, image1)

denom = math.sqrt(np.sum(image0 * image0) * np.sum(image1 * image1))
return np.sum(image0 * image1) / denom
```

where an import of the math module will have to be done at the top.

Please also make similar stylistic changes to how ROIs are handled in other functions.